packages/fuels-core/src/types/param_types.rs

```diff
@@ -164,36 +159,68 @@ impl ParamType {
     }
 
     /// Compute the inner memory size of a containing heap type (`Bytes` or `Vec`s).
-    pub fn heap_inner_element_size(&self, top_level_type: bool) -> Option<usize> {
-        match &self {
-            ParamType::Vector(inner_param_type) => inner_param_type.compute_encoding_in_bytes(),
+    pub fn heap_inner_element_size(&self, top_level_type: bool) -> Result<Option<usize>> {
+        let heap_bytes_size = match &self {
+            ParamType::Vector(inner_param_type) => {
+                Some(inner_param_type.compute_encoding_in_bytes()?)
+            }
             // `Bytes` type is byte-packed in the VM, so it's the size of an u8
             ParamType::Bytes | ParamType::String => Some(std::mem::size_of::<u8>()),
-            ParamType::StringSlice if !top_level_type => ParamType::U8.compute_encoding_in_bytes(),
-            ParamType::RawSlice if !top_level_type => ParamType::U64.compute_encoding_in_bytes(),
+            ParamType::StringSlice if !top_level_type => {
+                Some(ParamType::U8.compute_encoding_in_bytes()?)
+            }
+            ParamType::RawSlice if !top_level_type => {
+                Some(ParamType::U64.compute_encoding_in_bytes()?)
+            }
             _ => None,
-        }
+        };
+        Ok(heap_bytes_size)
     }
 
     /// Calculates the number of bytes the VM expects this parameter to be encoded in.
-    pub fn compute_encoding_in_bytes(&self) -> Option<usize> {
+    pub fn compute_encoding_in_bytes(&self) -> Result<usize> {
```

Comment:
Maybe we can make this and the test a bit cleaner by adding a closure:

```rust
let overflow_error = || {
            error!(
                InvalidType,
                "Reached overflow while computing encoding size for {:?}",
                self
            )
        };
```

Then used it like this:

```rust
ParamType::Array(param, count) => param
                .compute_encoding_in_bytes()?
                .checked_mul(*count)
                .ok_or_else(overflow_error),
```

If the `Debug` representation contains more info than needed we can go for `Display`. Maybe the `Display` implementation will also be useful in other places where we return `InvalidType`.